tsdfileapi/api.py

```diff
@@ -980,8 +980,8 @@ def prepare(self):
                                 try:
                                     if self.group_config['enabled']:
                                         subprocess.call(['chmod', '2770', target])
-                                        # todo: if not root dir, should be user not api_user
-                                        subprocess.call(['sudo', 'chown', f'{options.api_user}:{self.group_name}', target])
+                                        owner = options.api_user if target.endswith(self.group_name) else self.requestor
```

Comment:
What if the user uploads a target subfolder with the `group_name`? I'm not sure how likely it is that someone will do this, but is there a safer way to verify the intended case?

Reply:
good point, I'll use a different approach

Reply:
right, changed now

Reply:
Neat!